confmap/resolver.go

```diff
@@ -149,17 +147,15 @@ func (mr *Resolver) Resolve(ctx context.Context) (*Conf, error) {
 		}
 	}
 
-	if mr.enableExpand {
-		cfgMap := make(map[string]interface{})
-		for _, k := range retMap.AllKeys() {
-			val, err := mr.expandValueRecursively(ctx, retMap.Get(k))
-			if err != nil {
-				return nil, err
-			}
-			cfgMap[k] = val
+	cfgMap := make(map[string]interface{})
+	for _, k := range retMap.AllKeys() {
+		val, err := mr.expandValueRecursively(ctx, retMap.Get(k))
```

Comment:
Shouldn't we remove `expandconverter` now? At least from `newDefaultConfigProviderSettings`?

Reply:
No, read this does not expand unless a scheme is provided, and that is not the case for the `expandconverter`.